ebpf/xdp_synproxy.c

```diff
@@ -0,0 +1,283 @@
+// SPDX-License-Identifier: LGPL-2.1 OR BSD-2-Clause
```

Comment:
where does this come from? Did you modify it?

Reply:
this is from the Linux kernel tree bpf selftests https://elixir.bootlin.com/linux/latest/source/tools/testing/selftests/bpf/xdp_synproxy.c, I did modify it by removing some code from it.